app/main/views/suppliers.py

```diff
@@ -536,12 +557,56 @@ def invite_user(supplier_code):
             "view_supplier_users.html",
             status_code=400,
             invite_form=invite_form,
+            new_seller_user_form=new_seller_user_form,
             move_user_form=MoveUserForm(),
             users=users["users"],
             supplier=supplier
         )
 
 
+@main.route('/suppliers/<int:supplier_code>/add-new-supplier-user', methods=['POST'])
+@login_required
+@role_required('admin')
+def add_new_supplier_user(supplier_code):
```

Comment:
Let's try and use something similar to the seller dashboard invite feature.  You can update the existing invite user form.